packages/runtime/src/integrations.ts

```diff
@@ -108,6 +118,21 @@ export function createIntegration<Context extends RuntimeContext = RuntimeContex
                 return resp;
             }
 
+            if (
+                event.type === 'fetch_visitor_authentication' &&
+                definition.fetch_visitor_authentication
+            ) {
+                const resp = await definition.fetch_visitor_authentication(event, context);
+
+                logger.debug(
+                    `response ${resp.status} ${resp.statusText} Content-Type: ${resp.headers.get(
+                        'content-type'
+                    )}`
+                );
+
+                return resp;
+            }
+
```

Comment:
This should be done in a separate PR with a changeset for `runtime` package so it can be published separately